auth/scoperesolver.js

```diff
@@ -252,11 +252,12 @@ class ScopeResolver extends events.EventEmitter {
   }
 
   /**
-   * Return set of scopes that `scopes` can be expanded to when assuming all
-   * authorized roles.
+   * Return a normalized set of scopes that `scopes` can be expanded to when
+   * assuming all authorized roles.
    */
+
   resolve(scopes) {
-    let granted = dfa.sortScopesForMerge(_.clone(scopes));
+    let granted = dfa.sortScopesForMerge(ScopeResolver.normalizeScopes(scopes));
```

Comment:
I'm saying this because we have spent time making sure this code path was fast.. :)